pkg/types/types.go

```diff
@@ -74,7 +74,7 @@ type ResourcePool interface {
 type IBaseResource interface {
 	GetDeviceFile(dev string) (devFile string, err error)
 	GetDeviceSpecs(deviceFiles map[string]string, deviceIDs []string) []*pluginapi.DeviceSpec
-	GetEnvs(resourceName string, deviceIDs []string) map[string]string
```

Comment:
The IBaseResource interface is there so that the device specific information can be handled by some concrete resource pool types. Even though current implementation of netdevice,uio and vfio kind of send similar information, it gives more flexibility in future should we want to send more customized env var info. For example, we might want to ad _UIO" in the env var name or in the value.

To answer other question, The ResourcePool type is composition of some common methods(served by pool_stub.go) and IBaseResource interface. So when instantiate a concrete resourcPool each needs to implement its own methods for that interface. By embedding the stub resourcePool in a concrete resource allows access to its members directly.  Otherwise we need to get this information through parameters and method calls. (See how this was done before this changes)

Inheritance and polymorphism is not straight forward in Go.

Reply:
Yep, it's reasonably if we want to leave some room for further adjustment of env var per different types of resource.